src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -137,6 +157,16 @@ void initializeReference() {
      */
     void initializeReads() {
         if (! readArguments.getReadFiles().isEmpty()) {
+            // Prefetcher is useful for cloud files because of the latencies involved.
+            Function<SeekableByteChannel, SeekableByteChannel> wrapper = Function.identity();
+            Function<SeekableByteChannel, SeekableByteChannel> indexWrapper = Function.identity();
+
+            if (cloudPrefetchBuffer > 0) {
+                wrapper = is -> GATKTool.addPrefetcher(cloudPrefetchBuffer, is);
+            }
+            if (cloudIndexPrefetchBuffer > 0) {
+                indexWrapper = is -> GATKTool.addPrefetcher(cloudIndexPrefetchBuffer, is);
+            }
```

Comment:
This would make it little bit harder to experiment with different types of wrapper (like adding one that caches more but does less prefetching). Are you sure? There is no other usage of the ReadsDataSource constructor aside from tests so there aren't many clients to benefit from that simplification.